app/src/highlighter/index.ts

```diff
@@ -0,0 +1,191 @@
+
+extensionMIMEMap.set('.ts', 'text/typescript')
+extensionMIMEMap.set('.js', 'text/javascript')
+extensionMIMEMap.set('.json', 'application/json')
+
+import 'codemirror/mode/jsx/jsx'
+extensionMIMEMap.set('.tsx', 'text/typescript-jsx')
+extensionMIMEMap.set('.jsx', 'text/jsx')
+
+import 'codemirror/mode/htmlmixed/htmlmixed'
+extensionMIMEMap.set('.html', 'text/html')
+extensionMIMEMap.set('.htm', 'text/html')
+
+import 'codemirror/mode/css/css'
+extensionMIMEMap.set('.css', 'text/css')
+extensionMIMEMap.set('.scss', 'text/x-scss')
+extensionMIMEMap.set('.less', 'text/x-less')
+
+import 'codemirror/mode/markdown/markdown'
+extensionMIMEMap.set('.markdown', 'text/x-markdown')
+extensionMIMEMap.set('.md', 'text/x-markdown')
+
+import 'codemirror/mode/yaml/yaml'
+extensionMIMEMap.set('.yaml', 'text/yaml')
+extensionMIMEMap.set('.yml', 'text/yaml')
+
+import 'codemirror/mode/xml/xml'
+extensionMIMEMap.set('.xml', 'text/xml')
+extensionMIMEMap.set('.xaml', 'text/xml')
+extensionMIMEMap.set('.csproj', 'text/xml')
+extensionMIMEMap.set('.svg', 'text/xml')
+
+import 'codemirror/mode/clike/clike'
+extensionMIMEMap.set('.objc', 'text/x-objectivec')
+extensionMIMEMap.set('.scala', 'text/x-scala')
+extensionMIMEMap.set('.sc', 'text/x-scala')
+extensionMIMEMap.set('.cs', 'text/x-csharp')
+extensionMIMEMap.set('.java', 'text/x-java')
+extensionMIMEMap.set('.c', 'text/x-c')
+extensionMIMEMap.set('.h', 'text/x-c')
+extensionMIMEMap.set('.cpp', 'text/x-c++src')
+
+import 'codemirror/mode/shell/shell'
+extensionMIMEMap.set('.sh', 'text/x-sh')
+
+import 'codemirror/mode/go/go'
+extensionMIMEMap.set('.go', 'text/x-go')
+
+import 'codemirror/mode/perl/perl'
+extensionMIMEMap.set('.pl', 'text/x-perl')
+
+import 'codemirror/mode/php/php'
+extensionMIMEMap.set('.php', 'text/x-php')
+
+import 'codemirror/mode/python/python'
+extensionMIMEMap.set('.py', 'text/x-python')
+
+import 'codemirror/mode/ruby/ruby'
+extensionMIMEMap.set('.rb', 'text/x-ruby')
+
+function guessMimeType(contents: string) {
+  if (contents.startsWith('<?xml')) {
+    return 'text/xml'
+  }
+
+  if (contents.startsWith('#!')) {
+    const m = /^#!.*?(ts-node|node|bash|sh|python(?:[\d.]+)?)\r?\n/g.exec(
+      contents
+    )
+
+    if (m) {
+      switch (m[1]) {
+        case 'ts-node':
+          return 'text/typescript'
+        case 'node':
+          return 'text/javascript'
+        case 'sh':
+        case 'bash':
+          return 'text/x-sh'
+        case 'perl':
+          return 'text/x-perl'
+      }
+
+      if (m[1].startsWith('python')) {
+        return 'text/x-python'
+      }
+    }
+  }
+
+  return null
+}
+
+onmessage = (ev: MessageEvent) => {
+  const request = ev.data as IHighlightRequest
+
+  const tabSize = request.tabSize || 4
+  const extension = request.extension
+  const contents = request.contents
+
+  const mimeType = extensionMIMEMap.get(extension) || guessMimeType(contents)
+
+  if (!mimeType) {
+    console.debug(`Could not determine mime type for highlighting`)
```

Comment:
It does, yeah. Since it's not a very helpful message I'm just gonna remove it for now. If we decide we need logging in the workers we can always marshal it out and use `log.*`